File: android/src/main/java/com/reactlibrary/MotionSplashModule.java
package com.reactlibrary;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class MotionSplashModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext reactContext;

    public MotionSplashModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "MotionSplash";
    }

    @ReactMethod
    public void animate(String animationType) {
        startAnimation(animationType);
    }

    private void startAnimation(String animationType) {
        MotionSplashAnimationType type;

        if (animationType != null) {
            switch (animationType) {
                case "RotateOut":
                    type = MotionSplashAnimationType.rotateOut;
                    break;
                case "PopAndZoomOut":
                    type = MotionSplashAnimationType.popAndZoomOut;
                    break;
                case "SqueezeAndZoomOut":
                    type = MotionSplashAnimationType.squeezeAndZoomOut;
                    break;
                case "SwingAndZoomOut":
                    type = MotionSplashAnimationType.swingAndZoomOut;
                    break;
                case "WobbleAndZoomOut":
                    type = MotionSplashAnimationType.wobbleAndZoomOut;
                    break;
                case "HeartBeat":
                    type = MotionSplashAnimationType.heartBeat;
                    break;
                default:
                    type = MotionSplashAnimationType.twitter;
            }
        } else {
            type = MotionSplashAnimationType.twitter;
        }

        MotionSplash.animate(type);
    }

    @ReactMethod
    public void useCustomColor(boolean isApplyForIconImage, int color) {
        MotionSplash.useCustomColor(isApplyForIconImage, color);
    }

    @ReactMethod
    public void setMinimumBeats(int minimumBeats) {
        MotionSplash.setMinimumBeats(minimumBeats);
    }

    @ReactMethod
    public void hide() {
        MotionSplash.hide();
    }
}

File: ios/MotionSplashModule.swift
//
//  MotionSplashModule.swift
//  MotionSplash
//
//  Created by Mai Trung Duc on 13/5/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import RevealingSplashView

@objc open class MotionSplashModule: NSObject {
    @objc open var revealingSplashView: RevealingSplashView
    
    @objc public init(iconImage: UIImage, iconInitialSize: CGSize, backgroundColor: UIColor) {
        revealingSplashView = RevealingSplashView(iconImage: iconImage,iconInitialSize: iconInitialSize, backgroundColor: backgroundColor)
        super.init()
    }
    
    @objc public init(iconImage: UIImage, iconInitialSize: CGSize, backgroundImage: UIImage) {
        revealingSplashView = RevealingSplashView(iconImage: iconImage,iconInitialSize: iconInitialSize, backgroundImage: backgroundImage)
        super.init()
    }
    
    // TODO: custom color is not working for background image
    @objc open func useCustomColor(applyForIconImage: Bool, color: UIColor) {
        revealingSplashView.useCustomIconColor = applyForIconImage
        revealingSplashView.iconColor = color
    }
    
    @objc open func animate() {
        revealingSplashView.startAnimation()
    }
    
    @objc open func timing(duration: Double, delay: Double) {
        revealingSplashView.duration = duration
        revealingSplashView.delay = delay
    }
    
    @objc open var type: String {
        set {
            switch(newValue) {
                case "RotateOut":
                    revealingSplashView.animationType = .rotateOut
                    break
                case "PopAndZoomOut":
                    revealingSplashView.animationType = .popAndZoomOut
                    break
                case "SqueezeAndZoomOut":
                    revealingSplashView.animationType = .squeezeAndZoomOut
                    break
                case "SwingAndZoomOut":
                    revealingSplashView.animationType = .swingAndZoomOut
                    break
                case "WobbleAndZoomOut":
                    revealingSplashView.animationType = .woobleAndZoomOut // TODO: review typo here
                    break
                case "HeartBeat":
                    revealingSplashView.animationType = .heartBeat
                    break
                default:
                    revealingSplashView.animationType = .twitter
            }
        }
        get {
            return ""
        }
    }
    
    @objc open func hide() {
        revealingSplashView.heartAttack = true
    }
    
    @objc open func setMinimumBeats(value: Int) {
        revealingSplashView.minimumBeats = value > 0 ? value : 1
    }
}

